src/python/test/test_dxclient.py

```diff
@@ -1419,6 +1419,15 @@ def test_dx_upload_mult_paths_with_dest(self):
                 listing = run("dx ls /destdir/a").split("\n")
                 self.assertIn(os.path.basename(fd2.name), listing)
 
+    def test_dx_upload_empty_file(self):
+        testdir = tempfile.mkdtemp()
+        os.mkdir(os.path.join(testdir, 'a'))
+        with testutil.TemporaryFile(dir=os.path.join(testdir, "a")) as fd:
+            fd.close()
+            with temporary_project("test_dx_upload_empty_file default", select=True) as upload_empty_file_project_default:
+                run("dx upload {}".format(fd.name))
```

Comment:
Done.

I created a new test in `src/python/test/test_dxpy.py`, but didn't remove this one (to be sure it works fine on all layers). Inform me, please, if testing that from the CLI level is not necessary - I would remove this test.